src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -135,14 +171,20 @@ object GenomeReference {
   def gen: Gen[GenomeReference] = for {
     name <- Gen.identifier
     nContigs <- Gen.choose(3, 50)
-    contigs <- Gen.distinctBuildableOfN[Array, Contig](nContigs, Contig.gen)
-    contigNames = contigs.map(_.name).toSet
-    xContig <- Gen.oneOfSeq(contigNames.toSeq)
-    yContig <- Gen.oneOfSeq((contigNames - xContig).toSeq)
-    mtContig <- Gen.oneOfSeq((contigNames - xContig - yContig).toSeq)
+    contigs <- Gen.distinctBuildableOfN[Array, String](nContigs, Gen.identifier)
+    lengths <- Gen.distinctBuildableOfN[Array, Int](nContigs, Gen.choose(1000000, 500000000))
+    xContig <- Gen.oneOfSeq(contigs)
+    yContig <- Gen.oneOfSeq((contigs.toSet - xContig).toSeq)
+    mtContig <- Gen.oneOfSeq((contigs.toSet - xContig - yContig).toSeq)
     parX <- Gen.distinctBuildableOfN[Array, Interval[Locus]](2, Interval.gen(Locus.gen(Seq(xContig))))
     parY <- Gen.distinctBuildableOfN[Array, Interval[Locus]](2, Interval.gen(Locus.gen(Seq(yContig))))
-  } yield GenomeReference(name, contigs, Set(xContig), Set(yContig), Set(mtContig), parX ++ parY)
+  } yield GenomeReference(name, contigs, contigs.zip(lengths).toMap, Set(xContig), Set(yContig), Set(mtContig), parX ++ parY)
```

Comment:
Just to be clear, are you suggesting that any contig from `contigs` should be allowed in `par` and not just those in `xContigs` and `yContigs`?